File: ValidateSQLInjection.cs
class IsSQLInjec
{
    private bool IsSQLInjection(params string[] Collection)
    {
        bool val = false;
        if (Collection.Length > 0)
        {
            for (int i = 0; i <= Information.UBound(Collection, 1); i++)
            {
                if (ValidateString(Collection[i]))
                {
                    val = true;
                    break;
                }
            }
        }
        else
            val = true;
        return val;
    }
    private bool IsSQLInjection(List<string> ls)
    {
        bool val = false;
        if (ls.Count > 0)
        {
            for (var i = 0; i <= (ls.Count - 1); i++)
            {
                if (ValidateString(ls[i]))
                {
                    val = true;
                    break;
                }
            }
        }
        else
            val = true;
        return val;
    }
    private bool IsSQLInjection(string param, bool isPath = false)
    {
        bool re = false;
        if (isPath)
        {
            if (System.IO.File.Exists(param))
                re = true;
        }
        else
            re = ValidateString(param);
        return re;
    }
    private bool ValidateString(string param)
    {
        bool isSQLInjection = false;
        string[] sqlCheckList = new[] { "--", ";--", ";", "/*", "*/", "@@", "@", "char", "nchar", "varchar", "nvarchar", "alter", "begin", "cast", "create", "cursor", "declare", "delete", "drop", "end", "exec", "execute", "fetch", "insert", "kill", "select", "sys", "sysobjects", "syscolumns", "table", "update" };
        string CheckString = param.Replace("'", "''");
        int i = 0;
        while (i <= sqlCheckList.Length - 1)
        {
            if ((CheckString.IndexOf(sqlCheckList[i], StringComparison.OrdinalIgnoreCase) >= 0))
                isSQLInjection = true;
            System.Math.Max(System.Threading.Interlocked.Increment(ref i), i - 1);
        }
        return isSQLInjection;
    }
}

File: ValidateSQLInjection.vb
#Region "Security"
    Private Function IsSQLInjection(ByVal ParamArray Collection() As String) As Boolean
        Dim val As Boolean = False
        If Collection.Length > 0 Then
            For i As Integer = 0 To UBound(Collection, 1)
                If ValidateString(Collection(i)) Then
                    val = True
                    Exit For
                End If
            Next i
        Else
            val = True
        End If
        Return val
    End Function
    Private Function IsSQLInjection(ByVal ls As List(Of String)) As Boolean
        Dim val As Boolean = False
        If ls.Count > 0 Then
            For i = 0 To (ls.Count - 1)
                If ValidateString(ls(i)) Then
                    val = True
                    Exit For
                End If
            Next
        Else
            val = True
        End If
        Return val
    End Function
    Private Function IsSQLInjection(ByVal param As String, Optional ByVal isPath As Boolean = False) As Boolean
        Dim re As Boolean = False
        If isPath Then
            If System.IO.File.Exists(param) Then
                re = True
            End If
        Else
            re = ValidateString(param)
        End If
        Return re
    End Function
    Private Function ValidateString(ByVal param As String) As Boolean
        Dim isSQLInjection As Boolean = False
        Dim sqlCheckList As String() = {"--", ";--", ";", "/*", "*/", "@@", _
          "@", "char", "nchar", "varchar", "nvarchar", "alter", _
          "begin", "cast", "create", "cursor", "declare", "delete", _
          "drop", "end", "exec", "execute", "fetch", "insert", _
          "kill", "select", "sys", "sysobjects", "syscolumns", "table", _
          "update"}
        Dim CheckString As String = param.Replace("'", "''")
        Dim i As Integer = 0
        While i <= sqlCheckList.Length - 1
            If (CheckString.IndexOf(sqlCheckList(i), StringComparison.OrdinalIgnoreCase) >= 0) Then
                isSQLInjection = True
            End If
            System.Math.Max(System.Threading.Interlocked.Increment(i), i - 1)
        End While
        Return isSQLInjection
    End Function
#End Region
